### ISSUE TITLE
[Question]: Using Version 1.17.11 with Java 17 & How to run multiple servers on 1 IP

### ISSUE THREAD

**Issue by dmcovell** [AllTheMods/ATM-6#2850]
### What would you like to know?

I am attempting to run a machine with multiple minecraft worlds on it. One as ATM6 1.17.11 & the other as Vanilla MC 1.18.1. I have two main questions: 

1. How to run ATM6 with a different Java version? 
2. How to alter the server properties to run two servers on one IP address.

For the first question, I saw a related post that talked about forcing ATM6 to use a different Java version than the native Java 11. However, I wasn't sure how to proceed as I am not too familiar with Ubuntu and all of this. 
Now, I know the first comment will be. Why run ATM6 on Java 17? That is because Minecraft 1.18 requires that version of Java or newer. I have already managed to update the version of Java to 'java 17.0.1 2021-10-19 LTS' and gotten the Vanilla server to run on its own. Prior to the java version update, I did have ATM6 running fine on this machine. So I should have a good base to work from.

For the second question, I have seen a few posts about running multiple minecraft servers on one IP that involved changing the port numbers. And opening a new port to forward in the router/modem.
I plan on making the 1.18.1 world port 25566, instead of the default 25565. What precisely would I need to change in the server.properties file? Just the server port? Or would I have to change the query port & rcon.port too?
And how would I tell users to put in the IP at their computers? Would it be like this? 111.222.3.44:**25566**

Thank you in advance, and I look forward to learning more about all of this.

**Comment by jeremiahwinsley** [comment]
You can set the Java path for ATM6 in `server-setup-config.yaml`, look for the `forcedJavaPath` property. ATM6 is compatible with any version between 8 and 15.

Port number would be the same as vanilla. You'd just need to change the server port, none of the other ports. Then you'd use ip:port to access it, like your example.

**Comment by dmcovell** [comment]
So I cannot run 1.18 and ATM6 on the same machine?

Is there a way I can run two versions of Java? So each server uses their needed Java versions? Can someone direct me to a tutorial, or some detailed instructions to help me do this?

**Comment by jeremiahwinsley** [comment]
Like I said, you just need to set the path in `server-setup-config.yaml` so that ATM6 uses a different version of Java. 1.18 requires Java 17.

**Comment by radj307** [comment]
You can run as many servers on the same machine as you want using as many java installations as you want, so long as each server is using different ports.

Set different ports by changing these lines in `server.properties`:
```ini
query.port=XXXXX
rcon.port=XXXXX
server-port=XXXXX
```
Specify the java version by changing this line in `server-setup-config.yaml`:
```yaml
  forcedJavaPath: "/usr/lib/jvm/adoptopenjdk-16-hotspot-jre-amd64/bin/java"
```

**Comment by dmcovell** [comment]
Thank you guys! I managed to get both versions of minecraft server running! 

For those curious I typed "/usr/lib/jvm/java-1.11.0-openjdk-amd64/bin/java" for the forced path line in the server-setup-config.yaml file while in Ubuntu 18.04 LTS. 

Use the "update-java-alternatives --list" command in the terminal to locate your particular path, but be sure to add "/bin/java" to the end of whatever that command gives you.
